from asyncio.log import logger
import traceback
from aiogram import Bot, Router, types

router = Router()

@router.error()
async def error_handler(event: types.ErrorEvent, bot: Bot):
    error = event.exception
    error_type = error.__class__.__name__
    error_message = str(event.update.message.text)
    
    # –ü–æ–ª—É—á–∞–µ–º —Å–æ–∫—Ä–∞—â–µ–Ω–Ω—ã–π traceback
    tb_list = traceback.format_exception(type(error), error, error.__traceback__)
    tb_short = "".join(tb_list[-2:])
    error_summary = (
        f"üö® –û—à–∏–±–∫–∞:\n"
        f"‚Ä¢ –¢–∏–ø: {error_type}\n"
        f"‚Ä¢ –°–æ–æ–±—â–µ–Ω–∏–µ: {error_message}\n"
        f"‚Ä¢ –ú–µ—Å—Ç–æ:\n{tb_short}"
    )
    print(error_summary)
    await bot.send_message(2098644058, error_summary)
    return True